x-pack/filebeat/input/entityanalytics/provider/azuread/conf.go

```diff
@@ -34,6 +36,11 @@ func (c *conf) Validate() error {
 	if c.UpdateInterval == 0 {
 		return errors.New("update_interval must not be zero")
 	}
+	switch strings.ToLower(c.Dataset) {
+	case "", "all", "users", "devices":
+	default:
+		return errors.New("dataset must be 'all', 'users', 'devices' or empty")
+	}
```

Comment:
https://github.com/elastic/beats/pull/36441/files#diff-0e5363fa11585b0c91afa955323a2a21bd84b553790f73978acd587b55c097a0R283 seems to skip the device/user change if `dataset` is not one of `""` `"all"` `"users"` `"devices"`

Do we still need to validate and throw an error otherwise?

Reply:
The skips there don't include one of {"users", "devices"}. It could be expressed there as the complement, so that we skip when we are "users" for device collection and skip when we are "devices" for user collection, but negations are always less clear.